junit-platform-launcher/src/main/java/org/junit/platform/launcher/TestPlan.java

```diff
@@ -82,7 +82,7 @@ public static TestPlan from(Collection<TestDescriptor> engineDescriptors) {
 		return testPlan;
 	}
 
-	private TestPlan(boolean containsTests) {
+	protected TestPlan(boolean containsTests) {
```

Comment:
Although it would technically not prevent subclassing, it's perhaps a good idea to document this (now protected) constructor as `@API(status = INTERNAL ...)`.